### ISSUE TITLE
Pipeline support?

### ISSUE THREAD

**Issue by kplindegaard** [uglide/qredisclient#35]
I have a set of Qt apps that constantly produce a lot of key-value type data that must be shared between all of them. They could be run on a single node, or they may be distributed in separate containers in a cluster. Linux and Windows support is required. The amount of data is quite intense. Approximately >10000 updates a second with a ~1k serialized blob payload, so they have to be done in bulk. After a prototype performance test using [py-redis](https://github.com/andymccurdy/redis-py), it seems [Redis' pipelining](https://redis.io/topics/pipelining) feature should give adequate bulk insert performance. 

AFAICT, there aren't that many cross-platform [Redis clients](https://redis.io/clients) for C or C++ out there, and since I am already trapped in Qt-land I don't want to also depend on e.g. Boost or something else just for this matter. Therefore this client lib is currently my preferred starting point.

Which brings me to my questions: Any thoughts on support for doing mass insertions? qredisclient has most of the plumbing already in place, but it's not quite there. I would be happy to contribute to it, though, because I suspect bulk operations is not part of [RDM's](https://redisdesktop.com/) roadmap...?

**Comment by kplindegaard** [comment]
Thanks @uglide, I'll check it out.

**Comment by kplindegaard** [comment]
Hello @uglide 
It took me a little while to get this going, but I believe I'm making decent progress now. 

I have a fair implementation working now, but it ended up a little differently from what I first thought was possible. As I mentioned initially, I really need bulk insert performance, but inserting ~10000 values into an HSET (which is basically my main use case) in the same command  was really buggy. The actual number of values entered into the hash varied a lot, so I had to ditch that idea and in stead enclose a bulk insert with MULTI+EXEC.

But doing that exposed me to the separation of commands and response parsing in the transporter classes. Basically I couldn't use the logic in AbstractTransporter for parsing responses containing mostly QUEUED status messages. The response from a pipeline commands should drop the initial OK and n*QUEUED statuses and just return the response from trailing EXEC operation.  That's at least as far as I know how [redis-py](https://github.com/andymccurdy/redis-py) does it, I think. What I ended up doing was to extend DefaultTransporter and override some of the command issuing logic and response parsing in AbstractTransporter.

Phew, this comment is growing and growing... It's time to wrap up what I did on my [feature/pipeline branch](https://github.com/kplindegaard/qredisclient/tree/feature/pipeline):

- A new transporter class handling proper response parsing for pipelines with MULTI+EXEC. 
- A somewhat naive extension of the Command to support pipelines (not very elegant)
- Modifications to the Connection class (i.e. a new constructor parameter) to make it pick the correct transporter.
- Implemented a few unit tests for pipelines with benchmarks.

I would be happy to submit a pull request if you are interested.
